frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx

```diff
@@ -35,9 +40,20 @@ const InstallCluster: React.FC<InstallClusterProps> = ({ match }) => {
   React.useEffect(() => {
     k8sGet(ClusterServiceVersionModel, appName, ns)
       .then((clusterServiceVersionObj) => {
-        setIsIndependent(
-          getAnnotations(clusterServiceVersionObj)[OCS_SUPPORT_ANNOTATION].includes('external'),
-        );
+        const isIndependentSupported = getAnnotations(clusterServiceVersionObj)[
+          OCS_SUPPORT_ANNOTATION
+        ].includes('external');
+        if (isIndependentSupported) {
+          setIsIndependent(true);
+          const { configMaps = [], secrets = [], storageClasses = [] } = getRequiredKeys(
+            clusterServiceVersionObj,
+          );
+          setIndependentReqdKeys({ configMaps, secrets, storageClasses });
+          setDownloadFile(
+            createDownloadFile(getAnnotations(clusterServiceVersionObj)?.['externalClusterScript']),
```

Comment:
Absence of this annotation `getAnnotations(clusterServiceVersionObj)?.['externalClusterScript']` should be checked and handled as well. Or it should be checked with `isIndependentSupported`.